03-rnaseq/clustering/clustering_heatmap_example.Rmd

```diff
@@ -0,0 +1,296 @@
+The data that we downloaded from refine.bio for this analysis is a dataset, with
+a total of 24 samples (obtained from 6 mice with 4 technical replicates for each
+mouse), containing RNA-sequencing results for Group 3 and 4 medulloblastoma
+orthotopic xenograft mice with and without digoxin treatment.
+
+You can download this dataset by going to the following link and clicking 
+`Download Now`: https://www.refine.bio/experiments/SRP150101/transcriptome-analysis-of-group-3-and-4-medulloblastoma-orthotopic-xenograft-mice-with-digoxin-treatment
+
+```{r Import data from .tsv files}
+# Read in metadata tsv file
+metadata <- readr::read_tsv(file.path(
+  "data", # Replace with path to your metadata file
+  "metadata_SRP150101.tsv" # Replace with the name of your metadata file
+)) %>%
+  dplyr::select(refinebio_accession_code, refinebio_title, refinebio_treatment) %>%
+  replace(is.na(.), "None") %>%
+  dplyr::mutate(Groups = dplyr::case_when(
+    refinebio_treatment == "no treatment" ~ "MP",
+    TRUE ~ "SP"
+    # Replace with group names relevant to your data
+  ),
+  refinebio_title_collapsed = gsub("(_L).*", "", refinebio_title))
+
+# Read in data tsv file
+df <- readr::read_tsv(file.path(
+  "data", # Replace with path to your data file
+  "SRP150101.tsv" # Replace with the name of your data file
+)) %>%
+  tibble::column_to_rownames("Gene")
+```
+
+Let's ensure that the metadata and data are in the same sample order. 
+
+```{r}
+# Make the data in the order of the metadata
+df <- df %>% dplyr::select(metadata$refinebio_accession_code)
+
+# Check if this is in the same order
+all.equal(colnames(df), metadata$refinebio_accession_code)
+```
+
+## 3) Choose genes of interest
+Although you may want to create a heatmap including all of the genes in the set,
+alternatively, the heatmap could be created using only genes of interest. 
+For this example, we will sort genes by variance, but there are many alternative
+criterion by which you may want to sort your genes <i>eg</i> fold change,
+t-statistic, membership to a particular gene ontology, so on. 
+
+```{r Choose genes}
+# Calculate the variance for each gene
+variances <- apply(df, 1, var)
+
+# Determine the upper quartile variance cutoff value
+upper_var <- quantile(variances, 0.75)
+
+# Subset the data choosing only genes whose variances are in the upper quartile
+df_by_var <- df %>%
+  dplyr::filter(variances > upper_var)
+```
+
+## 4) Create a DESeqDataset
+A `DESeqDataSet` object is required to carry out the normalization step that
+you will see later using a function from the `DESeq2` package. This chunk of
+code will turn the data.frame (or matrix) into a `DESeqDataSet` object.
+
+```{r Create a DESeqDataset}
+# The `DESeqDataSetFromMatrix` function needs the values to be converted to
+# integers
+## TODO: Link to explanation about `DESeqDataSet` in the RNA-seq header section
+df_by_var <- df_by_var %>%
+  # Mutate numeric variables to be integers
+  dplyr::mutate_if(is.numeric, as.integer)
```

Comment:
`as.integer()` always rounds down, so it should be avoided.
```suggestion
  dplyr::mutate_if(is.numeric, round)
```

Reply:
Gotcha 👍